### DISCUSSION TITLE
Getting Started ( brew tap mongodb/brew - brew install mongodb-community )

### DISCUSSION THREAD

**Discussion by Nelson00011** [Homebrew/discussions#4336] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0c1031f2fd71016f0f5df699caaed384487bfb16
Last commit: 8 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0fa96312875e6246224d198c4dd9f1c9c773ff52
Core tap last commit: 10 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/ec2-user/environment/SandBox/proj_backend/homebrew
HOMEBREW_REPOSITORY: /home/ec2-user/environment/SandBox/proj_backend/homebrew
HOMEBREW_CELLAR: /home/ec2-user/environment/SandBox/proj_backend/homebrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 1
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.8 => /home/ec2-user/environment/SandBox/proj_backend/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: single-core 64-bit haswell
Clang: N/A
Git: 2.39.2 => /bin/git
Curl: 7.88.1 => /bin/curl
Kernel: Linux 4.14.304-226.531.amzn2.x86_64 x86_64 GNU/Linux
OS: Unknown
Host glibc: 2.26
/usr/bin/gcc: 7.3.1
/usr/bin/ruby: 2.0.0
glibc: 2.35_1
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

AWS IDE EC2 is having issues getting the 
- brew install mongodb-community -
command to work. 

Would love to know how to fix this. The pre-fix issue.

**Comment by ZhongRuoyu** [comment]
> AWS IDE EC2 is having issues getting the `brew install mongodb-community` command to work.

As mentioned in https://github.com/mongodb/homebrew-brew/pull/174#issuecomment-1433316812, the formulae in the `mongodb/brew` tap do not support Linux. (You can see from the [`mongodb-community` formula](https://github.com/mongodb/homebrew-brew/blob/master/Formula/mongodb-community.rb) that both download URLs listed there are for macOS.) If you want to install MongoDB Community, please look for alternatives. For example, their [official website](https://www.mongodb.com/try/download/community).

> Would love to know how to fix this. The pre-fix issue.

If you mean the `brew doctor` output (which is not related to the `mongodb-community` issue): please use the installer, as mentioned on https://brew.sh/:

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

After the installation, you may want to check out your shell profile (usually that's `~/.bash_profile` or `~/.profile` for Bash) for `brew`'s shell environment configs, if you added them before. Replace that with `eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"`.

You may do `rm -rf /home/ec2-user/environment/SandBox/proj_backend/homebrew` to remove the previous installation. (Heads up: this deletes everything you previously installed with `brew`.)

**Comment by Nelson00011** [comment]
- I managed to get homebrew installed on my Amazon IDE EC2 (Homebrew 4.0.6
Homebrew/homebrew-core (git revision 317cd988861; last commit 2023-03-18)
via https://docs.brew.sh/Installation#alternative-installs.
- Why won't the  mongodb/brew   tap work if I have HomeBrew installed? is
there anyway to get community vs homebrew for an IDE?

**Reply by ZhongRuoyu** [reply]
> Why won't the `mongodb/brew` tap work if I have HomeBrew installed?

As I mentioned in my earlier comment, it is the decision of `mongodb/brew` tap's authors not to provide support for Linux. This has nothing to do with Homebrew. Quoting https://github.com/mongodb/homebrew-brew/pull/174#issuecomment-1433316812:

> We don't currently have plans to support installing MongoDB via homebrew on linux.

---

> is there anyway to get community vs homebrew for an IDE?

I am not sure what you mean by "get community vs homebrew for an IDE". But: this is not the right place to ask for MongoDB support. This is Homebrew's discussion board where Homebrew-related problems are discussed.

Please make use of online resources if what you want is to install MongoDB Community on Amazon EC2. For example, MongoDB's [manual](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-amazon/).

**Comment by Nelson00011** [comment]
I thought it worked for linux:
https://tinmankinetics.com/how-to-install-mongo-in-macos-linux-with-homebrew/

**Reply by ZhongRuoyu** [reply]
This is not true, unfortunately.

**Reply by SMillerDev** [reply]
None of those paths work on Linux so that post is clearly one big lie

**Comment by Nelson00011** [comment]
Since I already have homebrew downloaded into the AWS Cloud9 IDE (would it
be possible to somehow override the URL for "
brew tap mongodb/brew") or could you recommend another alternative to
homebrew?

**Reply by ZhongRuoyu** [reply]
> could you recommend another alternative to homebrew

I have already recommended MongoDB's manual ["Install MongoDB Community Edition on Amazon Linux"](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-amazon/), assuming you're using Amazon Linux (other options are listed in sidebar). Please do not request MongoDB support in Homebrew's discussion board.

**Comment by Nelson00011** [comment]
Thank you for your email & being so helpful,
I definitely remember how responsive & helpful you were!
- still new to coding (6-months in).
